mutagen/wave.py

```diff
@@ -0,0 +1,369 @@
+        except (InvalidChunk, KeyError):
+            raise ID3NoHeaderError("No ID3 chunk")
+
+    @convert_error(IOError, error)
+    @loadfile(writable=True)
+    def save(self, filething, v2_version=4, v23_sep='/', padding=None):
+        """Save ID3v2 data to the Wave/RIFF file"""
+
+        fileobj = filething.fileobj
+
+        wave_file = WaveFile(fileobj)
+
+        if u'id3 ' not in wave_file:
+            wave_file.insert_chunk(u'id3 ')
+
+        chunk = wave_file[u'id3 ']
+
+        try:
+            data = self._prepare_data(
+                fileobj, chunk.data_offset, chunk.data_size, v2_version,
+                v23_sep, padding)
+        except ID3Error as e:
+            reraise(error, e, sys.exc_info()[2])
+
+        new_size = len(data)
+        new_size += new_size % 2  # pad byte
+        assert new_size % 2 == 0
+        chunk.resize(new_size)
+        data += (new_size - len(data)) * b'\x00'
+        assert new_size == len(data)
+        chunk.write(data)
+
+    @loadfile(writable=True)
+    def delete(self, filething):
+        """Completely removes the ID3 chunk from the RIFF/WAVE file"""
+
+        fileobj = filething.fileobj
+
+        waveFile = WaveFile(fileobj)
+
+        if 'id3 ' in waveFile:
+            waveFile['id3 '].delete()
+        if 'ID3 ' in waveFile:
+            waveFile['ID3 '].delete()
+
+        self.clear()
+
+
+@convert_error(IOError, error)
+@loadfile(method=False, writable=True)
+def delete(filething):
+    """Completely removes the ID3 chunk from the AIFF file"""
+
+    try:
+        del RiffFile(filething.fileobj)[u'ID3']
+    except KeyError:
+        pass
+
+
+class WAVE(FileType):
+    """WAVE(filething)
+
+    A Waveform Audio File Format
+    (WAVE, or more commonly known as WAV due to its filename extension)
+
+    Arguments:
+        filething (filething)
+
+    Attributes:
+        tags (`mutagen.id3.ID3`)
+        info (`WaveStreamInfo`)
+    """
+
+    _mimes = ["audio/wav", "audio/wave"]
+
+    @staticmethod
+    def score(filename, fileobj, header):
+        filename = filename.lower()
+
+        return (header.startswith(b"RIFF") * 2 + endswith(filename, b".wav") +
+                endswith(filename, b".wave"))
+
+    def add_tags(self):
+        """Add an empty ID3 tag to the file."""
+        if self.tags is None:
+            self.tags = _WaveID3()
+        else:
+            raise error("an ID3 tag already exists")
+
+    @convert_error(IOError, error)
+    @loadfile()
+    def load(self, filething, **kwargs):
+        """Load stream and tag information from a file."""
+
+        fileobj = filething.fileobj
+
+        try:
+            self.tags = _WaveID3(fileobj, **kwargs)
+        except ID3NoHeaderError:
+            self.tags = None
+        except ID3Error as e:
+            raise error(e)
+        else:
+            self.tags.filename = self.filename
+
+        fileobj.seek(0, 0)
+        self.info = WaveStreamInfo(fileobj)
```

Comment:
Can we load WaveStreamInfo() before the ID3 tags? Might make it fail earlier in case of a wrong format.